lib/hal/impression/rpihal.ex

```diff
@@ -0,0 +1,319 @@
+    do: write_command(state, @tres, packed_resolution)
+
+  # Panel Setting
+  # 0b11000000 = Resolution select, 0b00 = 640x480, our panel is 0b11 = 600x448
+  # 0b00100000 = LUT selection, 0 = ext flash, 1 = registers, we use ext flash
+  # 0b00010000 = Ignore
+  # 0b00001000 = Gate scan direction, 0 = down, 1 = up (default)
+  # 0b00000100 = Source shift direction, 0 = left, 1 = right (default)
+  # 0b00000010 = DC-DC converter, 0 = off, 1 = on
+  # 0b00000001 = Soft reset, 0 = Reset, 1 = Normal (Default)
+  defp set_panel(state), do: write_command(state, @psr, [0b11101111, 0x08])
+
+  defp set_power(state),
+    do:
+      write_command(state, @pwr, [
+        Bitwise.bor(
+          Bitwise.bor(
+            Bitwise.bor(
+              # ??? - not documented in UC8159 datasheet
+              Bitwise.<<<(0x06, 3),
+              # SOURCE_INTERNAL_DC_DC
+              Bitwise.<<<(0x01, 2)
+            ),
+            # GATE_INTERNAL_DC_DC
+            Bitwise.<<<(0x01, 1)
+          ),
+          # LV_SOURCE_INTERNAL_DC_DC
+          0x01
+        ),
+        # VGx_20V
+        0x00,
+        # UC8159_7C
+        0x23,
+        # UC8159_7C
+        0x23
+      ])
+
+  # Set the PLL clock frequency to 50Hz
+  # 0b11000000 = Ignore
+  # 0b00111000 = M
+  # 0b00000111 = N
+  # PLL = 2MHz * (M / N)
+  # PLL = 2MHz * (7 / 4)
+  # PLL = 2,800,000 ???
+  defp set_pll_clock_frequency(state), do: write_command(state, 0x3C)
+
+  defp set_tse_register(state), do: write_command(state, 0x00)
+
+  defp set_vcom_data_interval_setting(state, border),
+    do: write_command(state, @cdi, [Bitwise.bor(Bitwise.<<<(@colors[border], 5), 0x17)])
+
+  defp set_gate_source_non_overlap_period(state), do: write_command(state, @tcon, 0x22)
+  defp disable_external_flash(state), do: write_command(state, @dam, 0x00)
+  defp set_pws_whatever_that_means(state), do: write_command(state, @pws, 0xAA)
+  defp power_off_sequence(state), do: write_command(state, @pfs, 0x00)
+  defp push_pixel_buffer(state, buffer), do: write_command(state, @dtm1, buffer)
+  defp pon(state), do: write_command(state, @pon)
+  defp drf(state), do: write_command(state, @drf)
+  defp pof(state), do: write_command(state, @pof)
+
+  #
+  # waiting
+  #
+
+  defp await_device(state) do
+    case read_busy(state) do
+      0 ->
+        sleep(state, 10)
+        await_device(state)
+
+      1 ->
+        state
+    end
+  end
+
+  #
+  # pipe-able wrappers
+  #
+
+  defp sleep(state, sleep_time) do
+    io_call(state, :handle_sleep, [sleep_time])
+    state
+  end
+
+  defp set_reset(state, value) do
+    io_call(state, :handle_reset, [value])
+    state
+  end
+
+  defp read_busy(state) do
+    io_call(state, :handle_read_busy)
+  end
+
+  defp write_command(state, command) do
+    io_call(state, :handle_command, [command])
+    state
+  end
+
+  defp write_command(state, command, data) do
+    io_call(state, :handle_command, [command, data])
+    state
+  end
```

Comment:
This is why we don't copy-paste stuff 😅

Reply:
These are all being used. Is there an outstanding issue here?